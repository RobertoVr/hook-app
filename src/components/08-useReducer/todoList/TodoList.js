import React from 'react'
import { TodoListItem } from './TodoListItem'

export const TodoList = ({ todos, handleDelete, handleToggle }) => {
    return (
        <>
            <ul className="list-group list-group-fush">
                {
                    todos.map((todo, idx) => (
                        <TodoListItem todo={todo} idx={idx} handleDelete={handleDelete} handleToggle={handleToggle} />
                    ))
                }
            </ul>
        </>
    )
}
