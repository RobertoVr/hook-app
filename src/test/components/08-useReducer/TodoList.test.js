import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { TodoList } from '../../../components/08-useReducer/todoList/TodoList';
import { demoTodos } from '../../fixtures/demoTodos';

describe('Pruebas en TodoList', () => {

    const handleDelete = jest.fn();
    const handleToggle = jest.fn();

    const wrapper = shallow(<TodoList
        todos={demoTodos}
        handleDelete={handleDelete}
        handleToggle={handleToggle}
    />)

    test('Debe de hacer render el componente', () => {
        expect(wrapper).toMatchSnapshot();
    });
    test('Debe de tener dos TodoListItem', () => {
        expect(wrapper.find('TodoListItem').length).toBe(demoTodos.length);
        expect(wrapper.find('TodoListItem').at(0).prop('handleDelete')).toEqual(expect.any(Function));
    });

});
