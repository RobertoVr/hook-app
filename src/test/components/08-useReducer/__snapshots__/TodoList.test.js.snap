// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en TodoList Debe de hacer render el componente 1`] = `
<Fragment>
  <ul
    className="list-group list-group-fush"
  >
    <TodoListItem
      handleDelete={[MockFunction]}
      handleToggle={[MockFunction]}
      idx={0}
      key="1"
      todo={
        Object {
          "desc": "Aprender react",
          "done": false,
          "id": 1,
        }
      }
    />
    <TodoListItem
      handleDelete={[MockFunction]}
      handleToggle={[MockFunction]}
      idx={1}
      key="2"
      todo={
        Object {
          "desc": "Aprender mongo",
          "done": false,
          "id": 2,
        }
      }
    />
  </ul>
</Fragment>
`;
